### PR TITLE
Add coverage for Infinispan cache extension

### PR DESCRIPTION
**Pull request by gtroitsk**
### Summary
Test coverage for QQE-832
TP: https://github.com/quarkus-qe/quarkus-test-plans/blob/main/QQE-832.md 

New tests:
- Use [custom POJOs](https://quarkus.io/guides/cache-infinispan-reference#marshalling-your-pojos) as cache value, verify marshalling in Quarkus Infinispan. See `ExpensiveResponse` in cache service.
-  Test [Cache expiration](https://quarkus.io/guides/cache-infinispan-reference#expiration)  properties. `testCacheLifespan` and `testCacheMaxIdle`.

Please select the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Dependency update
- [ ] Refactoring
- [ ] Backport
- [x] New scenario (non-breaking change which adds functionality)
- [x] This change requires a documentation update
- [x] This change requires execution against OCP (use `run tests` phrase in comment)

### Checklist:
- [x] Methods and classes used in PR scenarios are meaningful
- [x] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

**Comment by gtroitsk** [comment]
run tests

**Comment by fedinskiy** [comment]
are there any test plans?

**Comment by gtroitsk** [comment]
> are there any test plans?

No test plan, if you require it I can create. The tests except caffeine cases must cover complex Java types marshalling and cache expiration. I went through [docs](https://quarkus.io/guides/cache-infinispan-reference) and [upstream PR](https://github.com/quarkusio/quarkus/pull/39836)

File: cache/infinispan/src/main/java/io/quarkus/ts/cache/infinispan/BaseServiceWithCache.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import io.quarkus.cache.CacheInvalidate;
+import io.quarkus.cache.CacheInvalidateAll;
+import io.quarkus.cache.CacheKey;
+import io.quarkus.cache.CacheResult;
+
+public abstract class BaseServiceWithCache {
+
+    private static final String CACHE_NAME = "service-cache";
+
+    private static int counter = 0;
```
Comment (fedinskiy) [inline cache/infinispan/src/main/java/io/quarkus/ts/cache/infinispan/BaseServiceWithCache.java:16]: It seems, that this value may potentially be changed from the several threads. Please, change this variable to Atomic or another race-protected alternative

**Review by fedinskiy — CHANGES_REQUESTED** [review]
Also, what about tracing and metrics? Does this extensions generate any?

**Comment by rsvoboda** [comment]
TP needs to be done for new features landing in product support

**Comment by gtroitsk** [comment]
run tests

**Review by fedinskiy — APPROVED** [review]
But please double-check, that failures in CI are unrelated

**Comment by gtroitsk** [comment]
Failures are unrelated for this PR, I work on them in the part of other task

**Comment by fedinskiy** [comment]
@gtroitsk could you link the task for a reference?

**Comment by gtroitsk** [comment]
> @gtroitsk could you link the task for a reference?

QQE-866

**Pull request merged**